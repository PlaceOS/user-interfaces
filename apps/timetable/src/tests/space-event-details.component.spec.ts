import { createRoutingFactory, SpectatorRouting } from '@ngneat/spectator/jest';

import { SpaceEventDetailsComponent } from '../app/space-event-details.component';

describe('SpaceEventDetailsComponent', () => {
    let spectator: SpectatorRouting<SpaceEventDetailsComponent>;
    const createComponent = createRoutingFactory(SpaceEventDetailsComponent);

    beforeEach(() => (spectator = createComponent()));

    it('should create component', () => {
        expect(spectator.component).toBeTruthy();
    });

    it('should match snapshot', () => {
        expect('[event]').not.toExist();
        spectator.setInput({
            event: { title: 'Event 1', date: 1, duration: 60 } as any,
        });
        spectator.detectChanges();
        expect('[event]').toExist();
    });
});
