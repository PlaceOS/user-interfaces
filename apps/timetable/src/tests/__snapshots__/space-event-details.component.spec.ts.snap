// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SpaceEventDetailsComponent should match snapshot 1`] = `
<div
  id="root1"
>
  <div
    class="absolute inset-x-1 bg-[#212121] text-white rounded border border-white/50 overflow-hidden px-2 py-1"
    style="top: 56.041666666666664%; height: 4.166666666666666%;"
  >
    <h2>
      
    </h2>
    <p>
        – 10:00 AM 
    </p>
  </div>
</div>
`;

exports[`SpaceEventDetailsComponent should match snapshot 2`] = `
<div
  id="root1"
>
  <div
    class="absolute inset-x-1 bg-[#212121] text-white rounded border border-white/50 overflow-hidden px-2 py-1"
    style="top: 41.66666666666667%; height: 4.166666666666666%;"
  >
    <h2>
      Event 1
    </h2>
    <p>
       10:00 AM – 11:00 AM 
    </p>
  </div>
</div>
`;
